src/components/Auth/Auth.tsx

```diff
@@ -1,44 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
 import { Navigate, Outlet, useLocation } from 'react-router-dom';
-import api from '../../api/api';
-
-const LOCALSTORAGE_USER_NAME = 'zaaktypenbeheer_user';
+import { api } from '../../api/api';
 
 export interface AuthContextType {
   user: any;
   signIn: (formData: FormData, callback: VoidFunction) => void;
   signOut: (callback: VoidFunction) => void;
 }
 
-let AuthContext = React.createContext<AuthContextType>(null!);
+const AuthContext = React.createContext<AuthContextType>(null!);
 
 export function AuthProvider({ children }: { children: React.ReactNode }) {
-  let [user, setUser] = React.useState<any>(localStorage.getItem(LOCALSTORAGE_USER_NAME));
+  const [authState, setAuthState] = useState<any>({
+    user: null,
+    authChecked: false,
+  });
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      const user = await api.getUser();
+      setAuthState({
+        user,
+        authChecked: true,
+      });
+    };
+    if (!authState.authChecked) fetchUser();
+  }, []);
```

Comment:
I think this can maybe be refactored to something along these lines:

```javascript
  // useAsync from react-use, so you can keep track if it's loading and then display a spinner or something.
  const { loading } = useAsync(() => {
    if (isLoggedIn) return;

    let userData;
    try {
       userData = await api.get(`${api.api_url}users/me/`);
    } catch (e) {
      // I think we should only catch the expected error (403). But any other unexpected case should bubble up
      if (!(e instanceof PermissionDenied)) throw e;
      return;
    }

    setUser(userData);
  }, []);
```

I think the `getUser` function can be removed from the `Auth.tsx` file since it is only used here.